clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerBackgroundThread.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.clients.consumer.internals;
+
+import org.apache.kafka.clients.consumer.internals.events.ConsumerRequestEvent;
+import org.apache.kafka.clients.consumer.internals.events.ConsumerResponseEvent;
+import org.apache.kafka.common.utils.KafkaThread;
+import org.apache.kafka.common.utils.Time;
+
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * The background thread runs in the background and consumes the {@code ApplicationEvent} from the {@link org.apache.kafka.clients.consumer.KafkaConsumer} APIs. This class uses an event loop to drive the following important tasks:
+ * <ul>
+ *  <li>Consuming and executing the {@code ApplicationEvent}.</li>
+ *  <li>Maintaining the connection to the coordinator.</li>
+ *  <li>Sending heartbeat.</li>
+ *  <li>Autocommitting.</li>
+ *  <li>Executing the rebalance flow.</li>
+ * </ul>
+ */
+public class ConsumerBackgroundThread extends KafkaThread implements AutoCloseable {
+    private static final String CONSUMER_BACKGROUND_THREAD_PREFIX = "consumer_background_thread";
+
+    private final BlockingQueue<ConsumerRequestEvent> consumerRequestEvent;
+    private final BlockingQueue<ConsumerResponseEvent> consumerResponseEvent;
+    private Time time;
+
+    // control variables
+    private volatile boolean closed = false;
+
+    public ConsumerBackgroundThread(BlockingQueue<ConsumerRequestEvent> consumerRequestEvents,
```

Comment:
The constuctor logic seems a bit reversed from what I'd expect. 

I would imagine the three-arg version would be be:

```
super(CONSUMER_BACKGROUND_THREAD_PREFIX, true);
this.time = time;
this.consumerRequestEvent = consumerRequestEvents;
this.consumerResponseEvent = consumerResponseEvents;
```

And then the two-arg version simply be:

```
this(Time.SYSTEM, consumerRequestEvents, consumerResponseEvents);
```